extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts

```diff
@@ -0,0 +1,367 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import { ClusterController, ControllerError } from '../controller/clusterControllerApi';
+import { AuthType } from '../constants';
+
+const localize = nls.loadMessageBundle();
+
+const basicAuthDisplay = localize('basicAuthName', "Basic");
+const integratedAuthDisplay = localize('integratedAuthName', "Windows Authentication");
+
+function getAuthCategory(name: AuthType): azdata.CategoryValue {
+	if (name === 'basic') {
+		return { name: name, displayName: basicAuthDisplay };
+	}
+	return { name: name, displayName: integratedAuthDisplay };
+}
+
+/**
+ * Converts a comma-delimited set of key value pair credentials to a JSON object.
+ * This code is taken from the azdata implementation written in Python
+ */
+function convertCredsToJson(creds: string): {} {
```

Comment:
This method is gnarly, but I couldn't see a saner / simpler way to do this. If you know how to take a string `a=b,c=\,ignorecomma,d=e==` and convert to 
```json
{
    credentials: {
        "a": "b",
        "c": "\,ignorecomma",
        "d": "e=="
    }
}
```
let me know!

Reply:
Is it always 'c' with that weird case of having a ,?

Reply:
Something along these lines could work...
```
let lastKey = '';
pair.forEach((p) => {
    const i = p.indexOf('=');
	if(i<0) {
	    x[lastKey]=`${x[lastKey]},${p}`;
		return;
	}
	const key = p.substring(0, i);
	const value = p.substring(i+1);
    x[key] = value;
	lastKey = key;
});
```

Reply:
I'm not sure if this is better than the current implementation which catches trailing escaped symbols. Honestly I wish backtrace in regex was supported in all browsers (since the Python example uses this and it's simpler). But I will keep as-is since it's close to the other implementation written, and I'm not clear if yours is solving the missing issues.